src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,272 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean; // Controls the visibility state of the modal
+  onClose: () => void; // Callback function to close the modal
+  className?: string; // Optional className override for the element
+};
+
+export function WalletModal({ isOpen, onClose, className }: WalletModalProps) {
+  const modalRef = useRef<HTMLDivElement>(null);
+  const [shouldRender, setShouldRender] = useState(isOpen);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      setShouldRender(true);
+    }
+  }, [isOpen]);
+
+  const handleAnimationEnd = () => {
+    if (!isOpen) {
+      setShouldRender(false);
+    }
+  };
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleSmartWalletConnector = useCallback(() => {
+    // WalletPreference.SMART_WALLET
+    connect({ connector: connectors[0] });
+    onClose();
+  }, [connect, connectors, onClose]);
+
+  const handleCoinbaseWalletConnector = useCallback(() => {
+    // WalletPreference.EOA
+    connect({ connector: connectors[1] });
```

Comment:
this will throw an index out of error range if the user doesn't provide more than one connector